### ISSUE TITLE
Invalid character error on valid ServerName directive containing scheme://

### ISSUE THREAD

**Issue by Sp1l** [icing/mod_md#321]
We're running into an error for domains where the [`ServerName` contains a `scheme://` prefix](https://httpd.apache.org/docs/2.4/mod/core.html#servername).

For servers where there's an SSL off-load in front of the Apache server, you must add scheme:// to the ServerName to force correct generation of redirect URLs.

```
incomplete: certificate(rsa) is missing        LetsEncrypt    on        Error[Missing parameter for the specified command line option]: urn:ietf:params:acme:error:rejectedIdentifier Error creating new order :: Cannot issue for "https://origin.www.example.com": Domain name contains an invalid character Next run in ~17 hours
```

**Comment by Sp1l** [comment]
> Oh, did not even know that was a thing.

Neither did I until I ran into the failing redirects 😃 

> Is this in the base server or in virtual hosts contexts?

This is in a VirtualHost context

> Would a ServerAlias with just the DNS name help as a temporary workaround?

We've just removed the `scheme://` prefix to make it work, our newer installs don't use SSL offloading. Created this issue more as a future improvement.

For completeness, we were using:

```
Define vhost somevhost.subdom.example.com

MDomain ${vhost}

<VirtualHost *:443>
ServerName  https://${vhost}

...

</VirtualHost>
```

**Comment by icing** [comment]
Thanks. I read httpd's code here:

* ServerName supports setting the scheme. If you set `https://${vhost}`, it correctly splits this into internal vars `server_scheme` and `server_hostname`. The latter is used by `mod_md` for matching and everything works.
* ServerAlias does **not** support this. It just take in the string configured. This means SNI will  not match here when you configure `ServerAlias https://origin.${vhost}`. And `mod_md` will not work also, as Lets Encrypt rejects this correctly as invalid DNS name.

tl;dr

What *should* work in your setup is:

```
<VirtualHost *:443>
ServerName  https://${vhost}
ServerAlias origin.${vhost}
...
</VirtualHost>
```

The `server_scheme` extracted from `ServerName` is also used for alias matches.

Therefore, I do not see anything to fix in `mod_md` regarding this. Do you agree?

**Comment by Sp1l** [comment]
Agree!

(sorry for the delay). Something to document?
